fractions/jose/src/main/java/org/wildfly/swarm/jose/Jose.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2015-2016 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.swarm.jose;
+
+/**
+ * Supports the protection of data with JOSE Signature and Encryption algorithms
+ */
+public interface Jose {
+    /**
+     * Sign the data in the JWS Compact or JSON (optional) format.
+     * @param data the data to be signed
+     * @return the signed data in the JWS Compact or JSON format
+     * @throws JoseException
+     */
+    String sign(String data) throws JoseException;
+
+    /**
+     * Sign the data in the JWS Compact or JSON (optional) format.
+     * @param input the data and optional JWS headers which have to be integrity-protected
+     * @return the signed data in the JWS Compact or JSON format
+     * @throws JoseException
+     */
+    String sign(SignatureInput input) throws JoseException;
+
+    /**
+     * Verify the signed data in the JWS compact or JSON (optional) format.
+     * @param jws the JWS sequence.
+     * @return verified data
+     * @throws JoseException
+     */
+    String verify(String jws) throws JoseException;
+
+    /**
+     * Verify the signed data in the JWS compact or JSON (optional) format.
+     * @param jws the JWS sequence.
+     * @return verified data and metadata
+     * @throws JoseException
+     */
+    VerificationOutput verification(String jws) throws JoseException;
```

Comment:
Should be called `verify` for consistency?

Reply:
Compiler does not like it.

Reply:
Ah of course! Sorry for the noise :-)

Reply:
Np :-)